src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -258,126 +268,78 @@ class PeerActor(
       context stop self
   }
 
+  //FIXME: As the peer chain check is part of the handshaker, it should be incorporated into the handshake
```

Comment:
So we would leave this `handlePeerChainCheck` in the `PeerActor` instead of it going in the `Handshaker`? I mainly thought about moving it there as the idea for our final design would be completely decoupling the `Ethereum` messages from the `PeerActor`, and maybe add a configurable message decoder if needed (as mentioned in https://github.com/input-output-hk/etc-client/pull/186/files#diff-d88dc0e9076bb9b1c58875187ebcd4acR407).